hack/generator/pkg/astmodel/object_function.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"github.com/dave/dst"
+)
+
+// objectFunction is a simple helper that implements the Function interface. It is intended for use for functions
+// that only need information about the object they are operating on
+// TODO: Reduce use of this in favour of the alternative in the functions package
```

Comment:
Ah, I was just coming back to ask about the difference between this one and the one in functions.

Reply:
Why not do that in this PR?

Seeing as structurally they seem identical it feels like a small reference update in a few usage locations? IIRC there are only 3 or 4 places this is used?

Reply:
It's far more than a few reference updates - the code consuming `astmodel.objectFunction` needs to be relocated out of `astmodel` before it can be modified to use `functions.ObjectFunction`. From discussion with @matthchr, this is possible but there may be fishhooks, so we agreed not to do it in this PR. I've created #1775 to track the required changes.